tools/pylib/_boutcore_build/boutcore.pyx.in

```diff
@@ -1137,21 +1606,26 @@ def bracket(Field3D a, Field3D b, method="BRACKET_STD", outloc="CELL_DEFAULT"):
 EOF
 for fun in sqrt exp sin cos log abs
 do
-    echo "def $fun(Field3D a):
+    echo "def $fun(a):
     \"\"\"
-    Calculate $fun of the Field3D
+    Calculate $fun of a
 
     Parameters
     ----------
-    a : Field3D
+    a : Field3D Field2D
          The field for which to calculate $fun
 
     Returns
     -------
-    Field3D
+    Field3D Field2D
         $fun of a
     \"\"\"
-    return f3dFromObj(c.$fun(a.cobj[0]))
+    if isinstance(a, Field3D):
+        return f3dFromObj(c.$fun((<Field3D?>a).cobj[0]))
+    elif isinstance(a, Field2D):
+        return f2dFromObj(c.$fun((<Field2D?>a).cobj[0]))
+    else:
+        raise NotImplemented(\"Unexpected argument type - not supported (yet?).\")
```

Comment:
Could this error include: 
 * what type was passed?
 * what types are supported?